src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -190,8 +193,8 @@ export class AccountManagementService implements IAccountManagementService {
 		});
 	}
 
-	private isCanceledResult(result: azdata.Account | azdata.PromptFailedResult): result is azdata.PromptFailedResult {
-		return (<azdata.PromptFailedResult>result).canceled;
+	private isPromptFailedResult(result: azdata.Account | azdata.PromptFailedResult): result is azdata.PromptFailedResult {
+		return typeof (<azdata.PromptFailedResult>result).canceled === 'boolean';
```

Comment:
Can you check type of result to be `azdata.Account` instead? and make this function `isResultAccount()`? 
We should be checking for valid results instead of failed result types if the goal is to know whether the result is success/failure.

Reply:
Given that it's currently only going to be one of these two types is there a specific reason why you think it should be reversed? The end result is the same.

Reply:
In this statement, we seem to be casting to `PromptFailedResult` directly, whereas what I'm proposing would check for type of result first.. but it's not critical, since it would not find `canceled` after casting.. I would also recommend to least check if result is `undefined`.